hathor/transaction/token_creation_tx.py

```diff
@@ -180,21 +182,29 @@ def serialize_token_info(self) -> bytes:
         encoded_symbol = self.token_symbol.encode('utf-8')
 
         ret = b''
-        ret += int_to_bytes(TOKEN_INFO_VERSION, 1)
+        ret += int_to_bytes(self.token_info_version, 1)
         ret += int_to_bytes(len(encoded_name), 1)
         ret += encoded_name
         ret += int_to_bytes(len(encoded_symbol), 1)
         ret += encoded_symbol
+
         return ret
 
     @classmethod
-    def deserialize_token_info(cls, buf: bytes, *, verbose: VerboseCallback = None) -> tuple[str, str, bytes]:
+    def deserialize_token_info(cls, settings: HathorSettings, buf: bytes, *, verbose: VerboseCallback = None) \
+            -> tuple[str, str, TokenInfoVersion, bytes]:
         """ Gets the token name and symbol from serialized format
         """
         (token_info_version,), buf = unpack('!B', buf)
         if verbose:
             verbose('token_info_version', token_info_version)
-        if token_info_version != TOKEN_INFO_VERSION:
+
+        try:
+            # blocks from deserialize with fee value when the feature is disabled
+            token_info_version = TokenInfoVersion(token_info_version)
+            if settings.FEE_FEATURE_FLAG is False and token_info_version != TokenInfoVersion.DEPOSIT:
+                raise ValueError()
```

Comment:
I don't get it. You are raising a `ValueError` here, but you are capturing it right after and then you raise it again. It seems you should raise something like a `FeeBasedTokensNotEnabled` exception. And this verification should be out of the try/except block.